hamp_radar/iquick.py

```diff
@@ -12,6 +12,35 @@
 from decoders import decode_srvi, decode_moment, decode_iq, decode_time
 
 
+def get_tag_size(data):
+    """
+    Extracts the tag (also known as signature) and the pointer for size
+    from 'data' assuming layout as in Meteorological Ka-Band Cloud Radar
+    MIRA35 Manual, section 2.3.1 'Definition of chunk common structure'
+
+    The tag is the first 4 bytes of data and the pointer for size is the
+    next 4 bytes. Bytes for tag are decoded to ascii string unless no valid
+    tag can be formed (see below). Bytes for size are interpreted as an integer
+
+    Args:
+        data (bytes): The data assumed to conform with Meteorological Ka-Band
+                      Cloud Radar MIRA35 Manual, section 2.3.1 'Definition of
+                      chunk common structure'
+
+    Returns:
+        Tuple[bytes, int]: The tag and the pointer for size.
+
+    Raises:
+        UnicodeDecodeError: If the first 4 bytes cannot be decoded as an ASCII string (e.g.
+        because a byte in the data for the tag represents a uint not in range 0-255), then
+        tag=None and size=0 are returned instead.
```

Comment:
Hmm... actually the function doesn't raise the UnicodeDecodeError, because we swallow it and return None instead...
So either we should make the function raise again, or add None, 0 to the possible return values.

Reply:
oh I didn't realise swallowed errors don't belong here, makes sense now I think about it... fixed now?

Reply:
I'd say you would want to describe what's visible from the outside, and this is: invalid ascii -> `None` is retruned. Users won't find out that there had been an exception raised when using the function.